blobstore/src/test/java/org/jclouds/blobstore/integration/internal/BaseBlobIntegrationTest.java

```diff
@@ -796,9 +808,8 @@ public void testCopyBlobReplaceMetadata() throws Exception {
          } finally {
             Closeables2.closeQuietly(is);
          }
-         // TODO: S3 overrideMetadataWith also overrides system metadata
-         // TODO: Swift does not preserve system metadata
-         //checkContentMetadata(toBlob);
+         // TODO: Swift copy object *appends* user metadata, does not overwrite
```

Comment:
You could query the source metadata before copy and apply that during the copy operation that overrides it. Otherwise you will have to do two operation that modify an object, instead of a read and a copy. Thoughts?